feat: bridge Roby data and pocolog data through task/port instances

doudou opened the pull request:
Depends on:
- [ ] https://github.com/rock-core/tools-roby/pull/241

This pull request improves data analysis by allowing to access "slices" of the data streams relevant for certain task instances, for instance: `roby.OroGen.someproject.Task.by_id(24).status_port` will now give access to the samples of the status_port stream relevant to task 24 from `someproject.Task.`

This is very practical for instance when trying to compare behavior of different instances of a given task, since we can iterate over the instances of with `each_task`.

doudou committed:
feat(dsl): define accessors for first and last emissions at model levels

doudou committed:
feat(dsl): define Root#each_task_model

doudou committed:
fix(dsl): event model for generated EventPropagation classes in TaskModel#each_event_propagation

doudou committed:
feat(log): memoize the loaded pocolog stream

doudou committed:
feat(log): bridge the Roby index with the data streams through ports

Accessing ports on a dataset's task instance gives access to the
corresponding data stream, restricted to the lifetime of the
underlying task.

This is compatible with e.g. samples_of in the DSL

doudou committed:
doc: accessors documentation

jhonasiv reviewed (approved):
I am not familiar with pocolog-syskit code, so all I did was try to check if there are any inconsistencies in the changes. Looking forward for these features when evaluating log data.

doudou committed:
fix: event propagation methods must return the corresponding accessor

The Dry entity should never "leak" to user code. Building useful accessors
requires tracking other values, and Dry "look like" hashes, which wreak
havoc in a lot of places.
